drag_and_drop_v2/public/js/drag_and_drop.js

```diff
@@ -41,6 +43,21 @@ function DragAndDropTemplates(configuration) {
         });
     };
 
+    var divideDragablesInTwoRows = function(dragables, itemsOrder) {
+        var dividedDragables = [];
+        var orderedDragables = [];
+        for (var i=0; i<itemsOrder.length; i++)
+        {
+            orderedDragables[itemsOrder[i]] = dragables[i];
+        }
+        for (var i = 0; i < orderedDragables.length; i=i+items_per_slide) {
+            var first_row = h("div.row", orderedDragables.slice(i, i+(items_per_slide/2)));
+            var second_row = h("div.row", orderedDragables.slice(i+(items_per_slide/2), i+items_per_slide));
+            dividedDragables.push(h("div.slide", [first_row, second_row]));
+        }
+        return dividedDragables;
+    };
+
```

Comment:
Instead of hardcoding the number of rows to 2, you can make that configurable as well, like this:
```diff
diff --git a/drag_and_drop_v2/public/js/drag_and_drop.js b/drag_and_drop_v2/public/js/drag_and_drop.js
index 35962b5..b56e84f 100644
--- a/drag_and_drop_v2/public/js/drag_and_drop.js
+++ b/drag_and_drop_v2/public/js/drag_and_drop.js
@@ -3,7 +3,8 @@ function DragAndDropTemplates(configuration) {

     var gettext;
     var ngettext;
-    var items_per_slide = 8;
+    var cols_per_slide = 2;
+    var rows_per_slide = 3;

     if ('DragAndDropI18N' in window) {
         // Use DnDv2's local translations
@@ -43,17 +44,21 @@ function DragAndDropTemplates(configuration) {
         });
     };

-    var divideDragablesInTwoRows = function(dragables, itemsOrder) {
+    var divideDragablesIntoRows = function(dragables, itemsOrder) {
         var dividedDragables = [];
         var orderedDragables = [];
         for (var i=0; i<itemsOrder.length; i++)
         {
             orderedDragables[itemsOrder[i]] = dragables[i];
         }
-        for (var i = 0; i < orderedDragables.length; i=i+items_per_slide) {
-            var first_row = h("div.row", orderedDragables.slice(i, i+(items_per_slide/2)));
-            var second_row = h("div.row", orderedDragables.slice(i+(items_per_slide/2), i+items_per_slide));
-            dividedDragables.push(h("div.slide", [first_row, second_row]));
+        var i = 0;
+        while (i < orderedDragables.length) {
+            var rows = [];
+            for (var j = 0; j < rows_per_slide; j=j+1) {
+                rows[j] = h("div.row", orderedDragables.slice(i, i+cols_per_slide));
+                i += cols_per_slide;
+            }
+            dividedDragables.push(h("div.slide", rows));
         }
         return dividedDragables;
     };
@@ -684,7 +689,7 @@ function DragAndDropTemplates(configuration) {
         bank_children = bank_children.concat(renderCollection(itemPlaceholderTemplate, items_placed, ctx));
         var itemsOrder = inBankItemsOrder.concat(placedItemsOrder);

-        bank_children = divideDragablesInTwoRows(bank_children, itemsOrder);
+        bank_children = divideDragablesIntoRows(bank_children, itemsOrder);
         var drag_container_style = {};
         var target_img_style = {};
         // If drag_container_max_width is null, we are going to measure the container width after this render.
```